src/main/java/de/tum/cit/aet/artemis/programming/service/localci/LocalCITriggerService.java

```diff
@@ -222,6 +228,8 @@ else if (triggeredByPushTo.equals(RepositoryType.TESTS)) {
         BuildJobQueueItem buildJobQueueItem = new BuildJobQueueItem(buildJobId, participation.getBuildPlanId(), buildAgent, participation.getId(), courseId,
                 programmingExercise.getId(), 0, priority, null, repositoryInfo, jobTimingInfo, buildConfig, null);
 
+        // Save the build job before adding it to the queue in case it got quickly processed. Update statement would then fail.
```

Comment:
I don't understand this comment. What do you mean with `Update statement would then fail`. Please explain this in more detail